components/engine_rocks/src/engine.rs

```diff
@@ -105,17 +106,24 @@ impl Iterable for RocksEngine {
 }
 
 impl Peekable for RocksEngine {
-    fn get_opt(&self, opts: &ReadOptions, key: &[u8]) -> Result<Option<Vec<u8>>> {
+    type DBVector = RocksDBVector;
+
+    fn get_value_opt(&self, opts: &ReadOptions, key: &[u8]) -> Result<Option<RocksDBVector>> {
         let opt: RocksReadOptions = opts.into();
         let v = self.0.get_opt(key, &opt.into_raw())?;
-        Ok(v.map(|v| v.to_vec()))
+        Ok(v.map(RocksDBVector::from_raw))
     }
 
-    fn get_cf_opt(&self, opts: &ReadOptions, cf: &str, key: &[u8]) -> Result<Option<Vec<u8>>> {
+    fn get_value_cf_opt(
+        &self,
+        opts: &ReadOptions,
+        cf: &str,
```

Comment:
Better to move `cf` ahead of opts? Keep consistent with `iterator_cf_opt`?

Reply:
+1

Reply:
There are a bunch of these situations that need to be consistent and putting cf in front of opts would be consistent with the `Mutable` trait, which currently looks like this:

```
pub trait Mutable {
    fn put_opt(&self, opts: &WriteOptions, key: &[u8], value: &[u8]) -> Result<()>;
    fn put_cf_opt(&self, opts: &WriteOptions, cf: &str, key: &[u8], value: &[u8]) -> Result<()>;

    fn delete_opt(&self, opts: &WriteOptions, key: &[u8]) -> Result<()>;
    fn delete_cf_opt(&self, opts: &WriteOptions, cf: &str, key: &[u8]) -> Result<()>;

    fn put(&self, key: &[u8], value: &[u8]) -> Result<()> {
        self.put_opt(&WriteOptions::default(), key, value)
    }

    fn put_cf(&self, cf: &str, key: &[u8], value: &[u8]) -> Result<()> {
        self.put_cf_opt(&WriteOptions::default(), cf, key, value)
    }

    fn delete(&self, key: &[u8]) -> Result<()> {
        self.delete_opt(&WriteOptions::default(), key)
    }

    fn delete_cf(&self, cf: &str, key: &[u8]) -> Result<()> {
        self.delete_cf_opt(&WriteOptions::default(), cf, key)
    }
    ...
}

When the `_opt` functions are viewed as refinements of the non-opt functions, shuffling the `cf` argument around to be in front of the introduced `opts` argument is inconsistent: `cf` and `key` always go together, so they shouldn't be split up by options.

If you want to rethink the order of arguments to these functions I would suggest either putting `opts` in front of all arguments or behind all arguments, but not in the middle.

I would be reluctant to change other functions not involved in this patchset though.